From be87630caba369e1681820d38cc3f3aaeaa18e49 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 19 Mar 2019 14:16:59 +0530
Subject: [PATCH 067/386] octeontx2-pf: Keep link notification enabled until
 device detach

Keep link notification enabled until device detach,
so that link notification broadcasting to VFs are not broken.

Change-Id: I9de07001b8d688e82132513302f0daa54a345657
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/6235
Tested-by: sa_ip-sw-jenkins
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 63df47792684..407020155e54 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1404,8 +1404,6 @@ int otx2_stop(struct net_device *netdev)
 	/* First stop packet Rx/Tx */
 	otx2_rxtx_enable(pf, false);
 
-	/* Disable link notifications */
-	otx2_cgx_config_linkevents(pf, false);
 
 	pf->intf_down = true;
 	/* 'intf_down' may be checked on any cpu */
@@ -1873,6 +1871,10 @@ static void otx2_remove(struct pci_dev *pdev)
 		return;
 
 	pf = netdev_priv(netdev);
+
+	/* Disable link notifications */
+	otx2_cgx_config_linkevents(pf, false);
+
 	unregister_netdev(netdev);
 	otx2_ptp_destroy(pf);
 	otx2_destroy_ethtool_flows(pf);
-- 
2.17.1

